projects/npm-tools/packages/npm-scripts/src/utils/createEsm2AmdIndexBridge.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * SPDX-FileCopyrightText: © 2019 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/* eslint-disable @liferay/no-dynamic-require */
+
+const fs = require('fs');
+const path = require('path');
+
+const getBndWebContextPath = require('./getBndWebContextPath');
+
+/**
+ * Create a .js file to make ES index module available as a Liferay-AMD module.
+ *
+ * @param {string} projectDir path to project's directory
+ * @param {object} buildConfig
+ */
+function createEsm2AmdIndexBridge(projectDir, buildConfig, manifest) {
+	const pkgJson = require(path.join(projectDir, 'package.json'));
+
+	const {output} = buildConfig;
+
+	const webContextPath = getBndWebContextPath() || `/${pkgJson.name}`;
+
+	const rootDir = pkgJson.name.startsWith('@') ? '../../../..' : '../../..';
```

Comment:
`'../../../..' : '../../..'` seems like a sort of "magic" destination. What do these two values reference?

Reply:
I will add a comment there because it's real magic. Those are the relative positions of each bridge related to the real ES module.

It is so strange because AMD modules are server under things like `/o/js/resolved-module/my-bundle$my-package@1.2.3/path/to/module.js` whereas ESM are under `/o/my-context-path/__liferay__/export/my-package.js`.

Also, depending on whether a package is npm-scoped or not, it needs to add/remove a `..`.